### PR TITLE
Issue-#132: Add possibility to change Cluster metadata dynamically after cluster started

### PR DESCRIPTION
**Pull request by antonkharenko**

### CODE REVIEW

File: cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java [hunk 1]
```diff
@@ -0,0 +1,235 @@
...
+    return messageObservable;
+  }
+
+  @Override
+  public void spreadGossip(Message message) {
+    gossip.spread(message);
+  }
+
+  @Override
+  public Observable<Message> listenGossips() {
```
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java]: no

File: cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java [hunk 2]
```diff
@@ -0,0 +1,235 @@
...
+import java.util.concurrent.ConcurrentMap;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Cluster implementation.
+ * 
+ * @author Anton Kharenko
+ */
+final class ClusterImpl implements Cluster {
```
Comment (antonkharenko) [inline cluster/src/main/java/io/scalecube/cluster/ClusterImpl.java]: this PR not related to it. Also I am not sure this is a good idea.

File: cluster/src/test/java/io/scalecube/cluster/ClusterTest.java [hunk 3]
```diff
@@ -29,9 +39,130 @@ public void testJoinDynamicPort() throws Exception {
...
+    for (Cluster node : otherNodes) {
+      node.listenMembership()
+          .filter(MembershipEvent::isUpdated)
+          .subscribe(event -> updateLatch.countDown());
+    }
+
+
+    // Update metadata
+    Map<String, String> updatedMetadata = ImmutableMap.of("key1", "value3");
+    metadataNode.updateMetadata(updatedMetadata);
```
Comment (antonkharenko) [inline cluster/src/test/java/io/scalecube/cluster/ClusterTest.java]: If you want to update only single property you need to use method updateMetadataProperty. This method will result in full replacement, so if you use updateMetadata then you need to provide all three properties.

Comment (ronenhamias) [inline cluster/src/test/java/io/scalecube/cluster/ClusterTest.java]: ok

**Pull request merged**